app/services/deployment/helpers/push_code_to_server.rb

```diff
@@ -20,6 +20,12 @@ def call
       private
 
       attr_reader :application_name
+
+      def clone_repo
+        # TODO: extract GithubClient so we can use any integration client
+        client = GithubClient.new(@configuration.installation_id)
+        GitWrapper.clone(@configuration.repo_path, client)
```

Comment:
It looks like `GitWrapper` already uses `git_client` as an initializer argument, may be it will be better to use instance method instead of class?

Reply:
I think it's misunderstanding - client here is `GithubClient`, `GitWrapper` uses "git" client. I'll rename this client to something more meaningful